### PR TITLE
DB/NPC: Quartermaster Bartlett Repair

### PR DESCRIPTION
**Pull request by poszer**
Fixed repair option for creature Quartermaster Bartlett in DragonBlight / Venomspite

##### CHANGES PROPOSED:

-  Updated gossip menu options
-  

###### ISSUES ADDRESSED:

Closes 

##### TESTS PERFORMED:

Tested in game, now it looks okay.
Before:
repairnooption
After: 
fixedrepair

##### HOW TO TEST THE CHANGES:

1. .go xyz 3209.35 -633.582 160.252 571
2. Talk with creature ( You will see option to Browse Goods & also repair items )

##### KNOWN ISSUES AND TODO LIST:

- [ ]
- [ ] 

##### Target branch(es):

Master

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1545780720578767600.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+INSERT INTO version_db_world (`sql_rev`) VALUES ('1545780720578767600');
+DELETE FROM `creature_template_addon` WHERE `entry`=27267;
+INSERT INTO `creature_template_addon` (`entry`, `bytes1`, `bytes2`, `emote`) VALUES (27267, 256, 257, 233);
+UPDATE `creature_template` SET `unit_flags`=32768 WHERE `entry`=27267;
+UPDATE `gossip_menu_option` SET `option_id`=3, `npc_option_npcflag`=128 WHERE `menu_id`=9487;
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1545780720578767600.sql]: Add an empty line at the end of the file.

**Comment by poszer** [comment]
> 
> 
> Tested. Repair option is now there and works.

Thank you for testings

**Comment by Stoabrogga** [comment]
In my opinion it would be better to change the SQL script as follows:
```
INSERT INTO version_db_world (`sql_rev`) VALUES ('1545780720578767600');

DELETE FROM `creature_template_addon` WHERE `entry` = 27267;
INSERT INTO `creature_template_addon` (`entry`, `bytes1`, `bytes2`, `emote`) VALUES (27267, 0, 0, 233);
UPDATE `gossip_menu_option` SET `option_id` = 3, `npc_option_npcflag` = 128 WHERE `menu_id` = 9487;
```
Explanation:
- Setting "bytes1" to 256 and "bytes2" to 257 in "creature_template_addon" is not necessary. Those values are also not documented in TC (see https://trinitycore.atlassian.net/wiki/spaces/tc/pages/2130001/creature+addon). It is the "emote"-value which counts here: 233 ("STATE_WORK_MINING")
- Setting "unit_flags" to 32768 in "creature_template" is also not necessary (32768 "UNIT_FLAG_UNK_15", see https://trinitycore.atlassian.net/wiki/spaces/tc/pages/2130008/creature+template).

I tested the new script and it is also working:
- "Mining"-animation is played for the NPC
- additional repair option is shown in the dialog

**Comment by poszer** [comment]
> In my opinion it would be better to change the SQL script as follows:
> 
> ```
> INSERT INTO version_db_world (`sql_rev`) VALUES ('1545780720578767600');
> 
> DELETE FROM `creature_template_addon` WHERE `entry` = 27267;
> INSERT INTO `creature_template_addon` (`entry`, `bytes1`, `bytes2`, `emote`) VALUES (27267, 0, 0, 233);
> UPDATE `gossip_menu_option` SET `option_id` = 3, `npc_option_npcflag` = 128 WHERE `menu_id` = 9487;
> ```
> Explanation:
> 
> * Setting "bytes1" to 256 and "bytes2" to 257 in "creature_template_addon" is not necessary. Those values are also not documented in TC (see https://trinitycore.atlassian.net/wiki/spaces/tc/pages/2130001/creature+addon). It is the "emote"-value which counts here: 233 ("STATE_WORK_MINING")
> * Setting "unit_flags" to 32768 in "creature_template" is also not necessary (32768 "UNIT_FLAG_UNK_15", see https://trinitycore.atlassian.net/wiki/spaces/tc/pages/2130008/creature+template).
> 
> I tested the new script and it is also working:
> 
> * "Mining"-animation is played for the NPC
> * additional repair option is shown in the dialog

I don't see any difference in the game if I apply your "suggestion". 
What is the point, since this already work fine? :thinking:

**Comment by Stoabrogga** [comment]
> I don't see any difference in the game if I apply your "suggestion".
> What is the point, since this already work fine?

To avoid unnecessary updates and inserts which could have unforeseen side effects. It is always good to work with the least amount of changes.

**Comment by poszer** [comment]
> 
> 
> > I don't see any difference in the game if I apply your "suggestion".
> > What is the point, since this already work fine?
> 
> To avoid unnecessary updates and inserts which could have unforeseen side effects. It is always good to work with the least amount of changes.

Okay, updated. Thanks

**Comment by BarbzYHOOL** [comment]
@poszer thank you for the PR!

@Stoabrogga thank you for review

@wetbrownsauce ty for testing

**Pull request merged**